WeatherDataFetcher.py

```diff
@@ -1,101 +1,166 @@
 import os
+import sys
 import datetime
 from typing import Tuple
 from geopy.geocoders import Nominatim
-import urllib.request, json
+import json
+import dotenv
+import requests
 
 
-class Weather_Forecast_Fetcher:
+def fetch_weather_data(api_key: str, location_coords: Tuple[int, int]) -> None:
     """
-    This class provides 3 days of forecast from an open API for a given location
-    api_key_open_weather_map.dat should contain the api key
+    Fetches weather data for 25 timepoints from the url if no recent data is available
+    Stores the data in the folder where it is executed for future use.
 
     Parameters:
-    ----------
-    city : stry
-      City for which the forcast will be given
-    location_coords : Tuple
-      Latitude and Longitude Coordinates of the location of interest (self.city)
-    api_key : str
-      API Key used for OpenWeatherMap
+    ---------
+    location_coords : Tuple[int:int]
+      (lat,long) coordinates for the city entered
+    recent_data_available : bool
+      if true, use cached values (load from file)
+      if false, reload from url (requests)
 
+    Returns: None
```

Comment:
I think the format is not right here. If it returns nothing we can skip it. Wdyt?

```suggestion
```